server/docs/design/block-persistence.md

```diff
@@ -3,64 +3,235 @@
 ## Table of Contents
 
 1. [Purpose](#purpose)
+1. [Terminology](#terminology)
+1. [Abstractions](#abstractions)
 1. [Goals](#goals)
-1. [Terms](#terms)
-1. [Entities](#entities)
-1. [Design](#design)
-
----
+1. [Overview](#overview)
+1. [Implementations](#implementations)
+   1. [Block as Local Directory (a.k.a. `block-as-local-dir`)](#block-as-local-directory-aka-block-as-local-dir)
+      1. [Overview](#overview-1)
+      1. [Specific implementations](#specific-implementations)
+      1. [Configurable Parameters](#configurable-parameters)
+   1. [Block as Local File (a.k.a. `block-as-local-file`)](#block-as-local-file-aka-block-as-local-file)
+      1. [Overview](#overview-2)
+      1. [Trie Structure & Algorithm for Block Path Resolution](#trie-structure--algorithm-for-block-path-resolution)
+      1. [Specific implementations](#specific-implementations-1)
+      1. [Configurable Parameters](#configurable-parameters-1)
+   1. [No Op (a.k.a. `no-op`)](#no-op-aka-no-op)
 
 ## Purpose
 
-The main objective of the `hedera-block-node` project is to replace the storage of Consensus Node artifacts (e.g.
-Blocks) on cloud storage buckets (e.g. GCS and S3) with a solution managed by the Block Node server. This document aims
-to describe the high-level design of how the Block Node persists and retrieves Blocks and how it handles exception cases
-when they arise.
+The main objective of the `hedera-block-node` project is to replace the storage
```

Comment:
Block Nodes are intended to serve many goals; replacing bucket storage is only one, and could not, by itself, justify creating the Block Nodes.
```suggestion
A major objective of the `hedera-block-node` project is to replace the storage
```

Reply:
✅ Done.